### PR TITLE
Use config_dir_path flag

### PR DESCRIPTION
**Pull request by ziishaned**

### CODE REVIEW

File: src/linter.ts [hunk 1]
```diff
@@ -9,10 +9,12 @@ export interface LinterError {
 }
 
 export default class Linter {
+  private workspaceFolder: any;
   private codeDocument: vscode.TextDocument;
 
   constructor(document: vscode.TextDocument) {
     this.codeDocument = document;
+    this.workspaceFolder = vscode.workspace.workspaceFolders[0];
```
Comment (jpreese) [inline src/linter.ts]: `workspaceFolders` doesn't appear to exist in the `workspace` namespace in my environment, though I do see _some_ usages of it in articles. Currently looking into where the gap is.

Comment (jpreese) [reply src/linter.ts]: Ah ok, it looks like we need to migrate away from the `vscode` dependency and leverage `@types/vscode` and outlined here https://code.visualstudio.com/api/working-with-extensions/testing-extension#migrating-from-vscode. 

We'll also need a check around if `vscode.workspace.workspaceFolders` is undefined before use to resolve lint error `TS2532`.

I did a quick POC and it now compiles with those changes.

[ ] Add @types/vscode to package.json
[ ] Add mocha to package.json
[ ] Remove vscode from package.json
[ ] Have the types and vscode.engines versions match
[ ] Add check around undefined workspaceFolders for TS2532

**Comment by jpreese** [comment]
- [ ] Add @types/vscode to package.json
- [ ] Add mocha to package.json
- [ ] Remove vscode from package.json
- [ ] Have the types and vscode.engines versions match
- [ ] Add check around undefined workspaceFolders for TS2532

File: src/linter.ts (hunk 1 above)
Comment (jpreese) [inline src/linter.ts]: If we want to leverage the `.workspaceFolders` member, we'll need to migrate away from the `vscode` dependency in `package.json` and use `@types/vscode`. (we should probably do this anyway, but right now without, this fails to compile)

A migration guide can be found here: https://code.visualstudio.com/api/working-with-extensions/testing-extension#migrating-from-vscode

**Comment by jpreese** [comment]
Could you also provide a use case for this? What problem are you running into? I'm concerned that a `workspace` is only for opening a folder, if you open a file, getting the path from the `workspace` isn't going to work. We'd need some extra logic around if a workspace exists or not.

That said, `protolint` should be finding the `.protolint.yaml` for you, even if the `.proto` is nested. Unless your use case has the protolint config elsewhere?

**Comment by jpreese** [comment]
Just checking in @ziishaned :)

**Comment by ziishaned** [comment]
@jpreese sorry for the late response. Currently this extension expects that you have a protolint configuration in the root. But in my case the configuration was in the project directory because of that the linting was not working.
Also I think its a good idea to use project configuration if it avaliable otherwise fall back to global configration(Users root directory)

**Comment by ziishaned** [comment]
I think we need to do some more changes regarding above usecase e.g. give the priority to the project directory configuration if its avaliable otherwise use global configuration.

**Comment by jpreese** [comment]
No worries, just wanted to get a clearer idea of what the problem was. If you've got an example folder structure where the extension doesn't pick up on the lint config, that'd be awesome.

Feel free to add some additional commits to your PR, otherwise I can try and whip something up tomorrow.

**Comment by jpreese** [comment]
The PR in its current form would still need some modifications, and seems a bit stale.

**Pull request closed without merging**